
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <title>LAMP | Data preparation</title>
    
    <meta name="author" content="Aika Terada">

    <!-- Le HTML5 shim, for IE6-8 support of HTML elements -->
    <!--[if lt IE 9]>
      <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->

    <!-- Le styles -->
    <link href="http://a-terada.github.com/lamp/assets/themes/twitter/bootstrap/css/bootstrap.min.css" rel="stylesheet">
    <link href="http://a-terada.github.com/lamp/assets/themes/twitter/css/style.css?body=1" rel="stylesheet" type="text/css" media="all">

    <!-- Le fav and touch icons -->
  <!-- Update these with your own images
    <link rel="shortcut icon" href="images/favicon.ico">
    <link rel="apple-touch-icon" href="images/apple-touch-icon.png">
    <link rel="apple-touch-icon" sizes="72x72" href="images/apple-touch-icon-72x72.png">
    <link rel="apple-touch-icon" sizes="114x114" href="images/apple-touch-icon-114x114.png">
  -->
  </head>

  <body>

    <div class="navbar">
      <div class="navbar-inner">
        <div class="container">
          <a class="brand" href="http://a-terada.github.com/lamp">LAMP</a>
          <ul class="nav">
            
            
            


  
    
      
      	<li><a href="http://a-terada.github.io/lamp/#installation">Installation</a></li>
      	<li><a href="http://a-terada.github.io/lamp/#demonstrations">Demonstrations</a></li>
      	<li><a href="http://a-terada.github.io/lamp/#usage">Usage</a></li>
      	<li><a href="#reference">Reference</a></li>
		<li class="navbar-text">|</li>
		<li class="active"><a href="http://a-terada.github.com/lamp/data_preparation.html" class="active">Data preparation</a></li>
      	
      
    
  
    
      
    
  



          </ul>
        </div>
      </div>
    </div>

    <div class="container">

      <div class="content">
        
<div class="page-header">
  <h1>Data preparation </h1>
</div>

<div class="row">
  <div class="span12">
    <!--## GEO and MSigDB data-->
<p>Here, we prepare codes for the analysis of real data. The LAMP takes a few hours to analyze the generated data.</p>

<p>These codes were written under the following assumptions:</p>

<ul>
<li><a href='http://www.broadinstitute.org/gsea/msigdb/index.jsp'>Molecular Signatures Database</a> (MSigDB) is used to generate the associations between the genes and the TFs.</li>

<li>The expression profile is observed by using microarray of Affymetrix.</li>
</ul>

<p>For an example, <a href='http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE6462'>GSE6462</a> dataset, which is stored in Gene Expression Omnibus (GEO), is used as the expression profile data. This example requires three files:</p>

<ul>
<li>
<dt>c3.tft.v3.0.entrez.gmt [<a href='http://www.broadinstitute.org/gsea/msigdb/download_file.jsp?filePath=/resources/msigdb/3.0/c3.tft.v3.0.entrez.gmt'>Download</a> (registration required)].</dt>

<dd>Associations between TF binding sites and genes which are collected in MSigDB. The motif gene set (c3) is used to give the associations between TFs and genes.</dd>
</li>

<li>
<dt>GSE6462_series_matrix.txt [<a href='ftp://ftp.ncbi.nlm.nih.gov/pub/geo/DATA/SeriesMatrix/GSE6462/GSE6462_series_matrix.txt.gz'>Download zip file</a>]</dt>

<dd>The gene expression profiling of human breast cancer cells.</dd>
</li>

<li>
<dt>HG-U133_Plus_2.na32.annot.csv [<a href='http://www.affymetrix.com/Auth/analysis/downloads/na32/ivt/HG-U133_Plus_2.na32.annot.csv.zip'>Download zip file</a> (registration required)]</dt>

<dd>The annotation file for Affymetrix Human Genome U133A 2.0 Array. The file is released by Affymetrix.</dd>
</li>
</ul>

<p>First, please download these files and decompress them in <code>lamp</code> directory. Then, try the following 6 commands:</p>

<pre><code>$ python convert_id.py HG-U133_Plus_2.na32.annot.csv GSE6462_series_matrix.txt to_exp.tab
$ python exp2ratio.py -t 12 to_exp.tab gse6462_expression.tab
$ python ratio2flag.py gse6462_expression.tab 1.0 gse6462_expression_over1.tab
$ python gsea4csv.py c3.tft.v3.0.entrez.gmt to_target.csv
$ python merge_exp_csv.py gse6462_expression.tab to_target.csv gse6462_target.csv
$ python check_files.py gse6462_target.csv gse6462_expression.tab</code></pre>

<p><code>gse6462_expression.tab</code>, <code>gse6462_expression_over1.tab</code> and <code>gse6462_target.csv</code> are generated by the above commands. The commands generate two other intermediate files: <code>to_exp.txt</code> and <code>to_target.csv</code>. The next figure illustrates the flow of the conversion files.</p>

<p><img src='images/flow.png' alt='Image not found' /></p>

<p>The flow is derived from the 6 codes:</p>

<ol>
<li>
<dl>
<dt>convert_id.py</dt>

<dd>Conversion of the prob ID to GenBank ID in Affymetrix microarray data.</dd>

<dd>Argument 1: The annotation file that is downloaded from Affymetrix.</dd>

<dd>Argument 2: GEO Series matrix file.</dd>

<dd>Argument 3: The output file (intermediate-file1)</dd>
</dl>
</li>

<li>
<dl>
<dt>exp2ratio.py</dt>

<dd>This code is written for the <code>GSE6462_series_matrix.txt</code>, which calculates target-condition&#8217;s log2 ratio of the expression levels to the control conditions for each gene.</dd>

<dd>Option -t: The target condition column number. The number starts 3 (column 0 is for the gene name, and columns 2 and 3 are control conditions).</dd>

<dd>Argument 1: The intermediate-file1.</dd>

<dd>Argument 2: Output file (expression-file when the Mann-Whitney U-test is selected to compute the p-value)</dd>
</dl>
</li>

<li>
<dl>
<dt>ratio2flag.py</dt>

<dd>Classification of the gene expression level to 1 (up-regulated) or 0 (unregulated). This code is available to generate the expression-file for the computation of p-value using Fisher&#8217;s exact test.</dd>

<dd>Argument 1: The expression-file for the Mann-Whitney U-test.</dd>

<dd>Argument 2: Threshold for division genes to 1 or 0.</dd>

<dd>Argument 3: Output file (expression-file when Fisher&#8217;s exact test is selected to compute the p-value)</dd>
</dl>
</li>

<li>
<dl>
<dt>gsea4csv.py</dt>

<dd>Changing the downloaded file from MSigDB to CSV format.</dd>

<dd>Argument 1: MSigDB collection file.</dd>

<dd>Argument 2: the output file (intermediate-file2).</dd>
</dl>
</li>

<li>
<dl>
<dt>merge_exp_csv.py</dt>

<dd>Addition/deletion of genes in the CSV formatting file to become the same gene set in expression-file. This code is available for any situation.</dd>

<dd>Argument 1: The expression-file.</dd>

<dd>Argument 2: CSV format file that contains the associations between the gene and TF.</dd>

<dd>Argument 3: The output file (target-file to LAMP)</dd>
</dl>
</li>

<li>
<dl>
<dt>check_files.py</dt>

<dd>Checking the gene set in both the target-file and the expression-file. When the two files contain different genes, these genes is printed. This code is available for any situation.</dd>

<dd>Argument 1: The target-file for LAMP.</dd>

<dd>Argument 2: The expression-file for LAMP.</dd>
</dl>
</li>
</ol>

<p>These codes are available for the other situations because the codes are run individually.</p>
  </div>
</div>


      </div>

      <footer>
        <p>&copy; Aika Terada 2012 
          with help from <a href="http://jekyllbootstrap.com" target="_blank" title="The Definitive Jekyll Blogging Framework">Jekyll Bootstrap</a>
          and <a href="http://twitter.github.com/bootstrap/" target="_blank">Twitter Bootstrap</a>
        </p>
      </footer>

    </div> <!-- /container -->

    


  <script type="text/javascript">
  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-37265974-1']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();
</script>



  </body>
</html>

