
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <title>HAT</title>
    
    <meta name="author" content="Aika Terada">

    <!-- Le HTML5 shim, for IE6-8 support of HTML elements -->
    <!--[if lt IE 9]>
      <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->

    <!-- Le styles -->
    <link href="assets/themes/twitter/bootstrap/css/bootstrap.min.css" rel="stylesheet">
    <link href="assets/themes/twitter/css/style.css?body=1" rel="stylesheet" type="text/css" media="all">

    <!-- Le fav and touch icons -->
  <!-- Update these with your own images
    <link rel="shortcut icon" href="images/favicon.ico">
    <link rel="apple-touch-icon" href="images/apple-touch-icon.png">
    <link rel="apple-touch-icon" sizes="72x72" href="images/apple-touch-icon-72x72.png">
    <link rel="apple-touch-icon" sizes="114x114" href="images/apple-touch-icon-114x114.png">
  -->
  </head>

  <body>

    <div class="navbar">
      <div class="navbar-inner">
        <div class="container">
          <a class="brand" href="/"></a>
          <ul class="nav">
            
            
            


  
    
      
    
  



          </ul>
        </div>
      </div>
    </div>

    <div class="container">

      <div class="content">
        
<div class="page-header">
  <h1>HAT  <small>High-Arity Transcription factor combination discovery</small></h1>
</div>

<div class="row">
  <div class="span12">
    <p>HAT is a code for multiple testing correction. Given relationships between transcription factors (TFs) and genes and expression levels of genes, the code list up statistically significant combinations of TFs. The HAT algorithm calibrates Family-Wise Error Rate (FWER) as the same level of Bonferroni correction. Applying our method to your data instead of Bonferroni correction gives you the below advantages:</p>

<ul>
<li>Listing up significant combinations without arity limit</li>

<li>Higher sensitivity</li>

<li>Faster calculation speed</li>
</ul>

<p>This manuscript is written for finding TF combinations. The code is, however, applicable to discover other combinatorial effects between other datasets such as miRNA interactions, histone acetylation and methylation and so on.</p>

<h2 id='preparation'>Preparation</h2>

<h3 id='requirements'>Requirements</h3>

<p>The programs are written in python (&gt;=2.6) except frequent pattern mining. As the mining program, <a href='http://research.nii.ac.jp/~uno/codes.htm'>Liner time Closed itemset Miner (LCM) ver. 5.3</a> was used. The installation way is written below.</p>

<h3 id='installation'>Installation</h3>

<p><strong>1. HAT</strong></p>

<p>If you have Git, you can clone the latest version by running:</p>

<pre><code>$ git clone https://github.com/a-terada/hat</code></pre>
<!-- "$" is NOT needed to type. It is a symbol to represent command line. -->
<p>When you do not have Git, you can download HAT in either <a href='https://github.com/a-terada/hat/zipball/master'>zip</a> or <a href='https://github.com/a-terada/hat/tarball/master'>tar</a> formats.</p>

<p><strong>2. LCM ver. 5.3</strong></p>

<p>As a frequent itemset mining program, HAT uses Liner time Closed itemset Miner (LCM) ver. 5.3 was used. It should be downloaded and compiled in the HAT directory before running of HAT by the following command:</p>

<pre><code>$ cd hat
$ wget &quot;http://research.nii.ac.jp/~uno/code/lcm53.zip&quot;
$ mkdir lcm53
$ tar zxvf lcm53.zip -C lcm53
$ pushd lcm53
$ make
$ popd</code></pre>

<h2 id='usage'>Usage</h2>

<p>In the HAT directory, you can run HAT with the following command:</p>

<pre><code>$ python multiple_test.py target-file expression-file significance-level</code></pre>

<p>HAT requires three arguments including two input file names and one one threshold:</p>

<ul>
<li>
<dl>
<dt><strong>target-file</strong></dt>

<dd>The file includes associations between TFs and genes.</dd>
</dl>
</li>

<li>
<dl>
<dt><strong>expression-file</strong></dt>

<dd>The file of gene expression.</dd>
</dl>
</li>

<li>
<dl>
<dt><strong>significance-level</strong></dt>

<dd>The statistical significance threshold.</dd>
</dl>
</li>
</ul>

<p>The input file and output format are written below.</p>

<p>The above command prints both result and running progress.</p>

<p>To find only result in your console, try:</p>

<pre><code>$ python multiple_test.py target-file expression-file significance-level 2&gt; /dev/null</code></pre>
<!-- “/dev/null” is a special file for disposing of unwanted output of a process. --><!-- Your operating system has already provides the file. -->
<p>To save the result to &#8216;output-file&#8217;, try:</p>

<pre><code>$ python hat.py target-file expression-file significance-level &gt; output-file</code></pre>

<p>To save the result to output-file and not to show the progress of computation, try:</p>

<pre><code>$ python hat.py target-file expression-file significance-level &gt; output-file 2&gt; /dev/null</code></pre>

<h2 id='input_file_format'>Input File Format</h2>

<p>HAT receives two input files: target-file and expression-file. Their examples are in the <code>sample/</code> directory.</p>

<h3 id='targetfile'>Target-file</h3>

<p>Target-file provides the associations between TFs and their target genes. This file should be formatted according in a CSV format. <code>sample/sample_target.csv</code> is an example.</p>

<pre><code>#gene,TF1,TF2,TF3,TF4
A,1,1,1,0
B,1,1,1,0
C,1,0,0,1
D,0,0,0,0
...</code></pre>

<p>This file must start with &#8217;#&#8217;. The line provides TFs names. Columns except the first indicate TF names used for the result.</p>

<p>The second-to-last line includes the associations between a gene and TFs. The first column is the gene name. The second-to-last column is whether the TF targets the gene. If the gene is targeted of the TF, then the column value is &#8216;1&#8217;. If not, the value is &#8216;0&#8217;. For instance, gene A is target gene of TF1, 2 and 3 in the example.</p>

<p><code>sample/sampe_target.csv</code> contains 15 genes and 4 TFs. Thus, the file composes with 16 lines and 5 columns.</p>

<h3 id='expressionfile'>Expression-file</h3>

<p>Expression-file provides gene expressions levels. The file is tab-delimited format with two columns. For example, we show the content of <code>samples/sample_expressions.tab</code> as the following:</p>

<pre><code>#gene	expression
A	1
B	1
C	0
D	0
...</code></pre>

<p>The comment line stars with &#8217;#&#8217; and is ignored from the analysis.</p>

<p>Each line contains two columns: a gene name and an expression level. HAT derives the p-value from Fisher&#8217;s exact test, thus the genes are classified two groups: up-regulated or unregulated. If the gene is up-regulated then the second column is 1, otherwise 0. In the above example, gene A and B are up-regulated, and gene C and D are unregulated.</p>

<p><code>sample/sampe_expression.tab</code> describes expressions of 15 genes.</p>

<h3 id='note_for_input_files'>Note for Input Files</h3>

<p>Gene orders in the target-file and the expression-file should be identical. For example, both <code>sample/sample_target.csv</code> and <code>sample/sample_expression.tab</code> consist of 15 genes from gene A to O. When the given gene sets are different, HAT does not work properly.</p>

<h2 id='output_format'>Output Format</h2>

<p>The result is printed out after &#8221;&#8212; results &#8212;&#8221; line. The first and second lines are formatted as the following:</p>

<pre><code>--- results ---
Threshold: [float], Correction factor: [integer] (# target genes &gt;= [integer])
# of significance: [integer]</code></pre>

<p><span>integer</span> and <span>float</span> are replaced with real number in the actual result. Each value means the following:</p>

<ul>
<li>
<dt><strong>Threshold</strong></dt>

<dd>The significance threshold.</dd>
</li>

<li>
<dt><strong>Correction factor</strong></dt>

<dd>The number of testable combinations. The adjusted p-value is computed with the multiplication of the raw p-value and this value.</dd>
</li>

<li>
<dt><strong># of target genes</strong></dt>

<dd>The minimum number of target genes targeted by a testable TF combination.</dd>
</li>

<li>
<dt><strong># of significant combinations:</strong></dt>

<dd>The number of significant combinations.</dd>
</li>
</ul>

<p>The detected combinations are shown after the third lines. Each line contains 5 columns:</p>

<ul>
<li>
<dt><strong>Raw p-value</strong></dt>

<dd>The calculated p-value with Fisher&#8217;s exact test.</dd>
</li>

<li>
<dt><strong>Adjusted p-value</strong></dt>

<dd>Adjusted p-value. The value of all combinations printed by HAT are smaller than the significance level. If the value is smaller than the inputted significance level, the combination becomes significant.</dd>
</li>

<li>
<dt><strong>Combination</strong></dt>

<dd>The significant combination. The TFs are delimited by &#8217;,&#8217; (comma).</dd>
</li>

<li>
<dt><strong># of target genes</strong></dt>

<dd>The number of target genes of the combination.</dd>
</li>

<li>
<dt><strong>Statistic score:</strong></dt>

<dd>The number of genes that both target and up-regulated.</dd>
</li>
</ul>

<p>The last line includes calculation time (in seconds):</p>

<pre><code>Time (sec.): Correction factor [float], P-value [float], Total [float]</code></pre>

<ul>
<li>
<dt><strong>Correction factor</strong></dt>

<dd>The time to calibrate the correction term.</dd>
</li>

<li>
<dt><strong>P-value</strong></dt>

<dd>The time to calculate p-values of testable combinations.</dd>
</li>

<li>
<dt><strong>Total</strong></dt>

<dd>Total time.</dd>
</li>
</ul>

<h2 id='demonstration'>Demonstration</h2>

<p>Two types of files are in in <code>sample/</code> directory. The target-file is sample_target.csv and expression-file is sample_expression.tab.</p>

<p>The dataset contains 4 TFs and 15 genes. The following command finds all significant combinations from the files with significance level &lt;= 0.05:</p>

<pre><code>$ python hat.py sample/sample_target.csv sample/sample_expression.tab 0.05 2&gt; /dev/null</code></pre>

<p>When the HAT finishes successfully, the result is printed in your console:</p>

<pre><code>--- results ---
Threshold: 0.01, Correction factor: 5 (# target genes &gt;= 5)
# of significance: 1
Raw p-value	Adjusted p-value	Combination	# target genes	Statistic score
0.00699300699301	0.034965034965	TF1,TF2,TF3	5	5.0
Time (sec.): Correction factor 0.005433, P-value 0.000172, Total 0.005605</code></pre>

<p>The result means that</p>

<ul>
<li>When the raw p-value &lt;= 0.01 (calculated by 0.05/5), the combination is called significant.</li>

<li>There is one significant combination. Ternary TFs TF1, 2 and 3 is significant. The raw p-value is 0.00699 and the adjusted p-value is 0.0350.</li>

<li>The running time is 0.00561 sec. in total.</li>
</ul>

<h2 id='note_on_the_running'>Note on the running</h2>

<p>HAT produces an LCM argument file and result files of LCM. One is saved as <code>target-file.4lcm53</code>, and the others are saved in <code>target-file.4lcm53.results/</code> directory. When these files exist, HAT reads LCM result files instead of running LCM.</p>

<p>Thus, if the HAT fail to finish the calculation, please command:</p>

<pre><code>$ rm -rf target-file.4lcm53.*</code></pre>

<p>LCM related files can be removed by the command. If you omit the commands, the calculation might fail again.</p>

<h2 id='input_file_format'>Input File Format</h2>

<p>HAT receives two input files: target-file and expression-file. The user should prepare the files.</p>

<h2 id='contact'>Contact</h2>

<p>Please contact the below address in case you find a bug, need detailed information or need application for different situation and so on.</p>

<ul>
<li>Aika Terada (terada_AT_ss.cs.titech.ac.jp)</li>
</ul>
  </div>
</div>


      </div>

      <footer>
        <p>&copy; Aika Terada 2012 
          with help from <a href="http://jekyllbootstrap.com" target="_blank" title="The Definitive Jekyll Blogging Framework">Jekyll Bootstrap</a>
          and <a href="http://twitter.github.com/bootstrap/" target="_blank">Twitter Bootstrap</a>
        </p>
      </footer>

    </div> <!-- /container -->

    
  </body>
</html>

